src/library-authoring/common/context/SidebarContext.tsx

```diff
@@ -0,0 +1,164 @@
+
+type SidebarProviderProps = {
+  children?: React.ReactNode;
+  /** Only used for testing */
+  initialSidebarComponentInfo?: SidebarComponentInfo;
+};
+
+/**
+ * React component to provide `LibraryContext`
+ */
+export const SidebarProvider = ({
+  children,
+  initialSidebarComponentInfo,
+}: SidebarProviderProps) => {
+  const [sidebarComponentInfo, setSidebarComponentInfo] = useState<SidebarComponentInfo | undefined>(
+    initialSidebarComponentInfo,
+  );
+
+  /** Helper function to consume addtional action once performed.
+    Required to redo the action.
+  */
+  const resetSidebarAdditionalActions = useCallback(() => {
+    setSidebarComponentInfo((prev) => (prev && { ...prev, additionalAction: undefined }));
+  }, []);
+
+  const closeLibrarySidebar = useCallback(() => {
+    setSidebarComponentInfo(undefined);
+  }, []);
+  const openAddContentSidebar = useCallback(() => {
+    setSidebarComponentInfo({ id: '', type: SidebarBodyComponentId.AddContent });
+  }, []);
+  const openInfoSidebar = useCallback(() => {
+    setSidebarComponentInfo({ id: '', type: SidebarBodyComponentId.Info });
+  }, []);
+
+  const openComponentInfoSidebar = useCallback((usageKey: string, additionalAction?: SidebarAdditionalActions) => {
+    setSidebarComponentInfo((prev) => ({
+      ...prev,
+      id: usageKey,
+      type: SidebarBodyComponentId.ComponentInfo,
+      additionalAction,
+    }));
+  }, []);
+
+  const openCollectionInfoSidebar = useCallback((
+    newCollectionId: string,
+    additionalAction?: SidebarAdditionalActions,
+  ) => {
+    setSidebarComponentInfo((prev) => ({
+      ...prev,
+      id: newCollectionId,
+      type: SidebarBodyComponentId.CollectionInfo,
+      additionalAction,
+    }));
+  }, []);
+
+  const setSidebarCurrentTab = useCallback((tab: CollectionInfoTab | ComponentInfoTab) => {
+    setSidebarComponentInfo((prev) => (prev && { ...prev, currentTab: tab }));
+  }, []);
+
+  const context = useMemo<SidebarContextData>(() => {
+    const contextValue = {
+      closeLibrarySidebar,
+      openAddContentSidebar,
+      openInfoSidebar,
+      openComponentInfoSidebar,
+      sidebarComponentInfo,
+      openCollectionInfoSidebar,
+      resetSidebarAdditionalActions,
+      setSidebarCurrentTab,
+    };
+
+    return contextValue;
+  }, [
+    closeLibrarySidebar,
+    openAddContentSidebar,
+    openInfoSidebar,
+    openComponentInfoSidebar,
+    sidebarComponentInfo,
+    openCollectionInfoSidebar,
+    resetSidebarAdditionalActions,
+    setSidebarCurrentTab,
+  ]);
+
+  return (
+    <SidebarContext.Provider value={context}>
+      {children}
+    </SidebarContext.Provider>
+  );
+};
+
+export function useSidebarContext(): SidebarContextData {
+  const ctx = useContext(SidebarContext);
+  if (ctx === undefined) {
+    /* istanbul ignore next */
+    throw new Error('useSidebarContext() was used in a component without a <SidebarProvider> ancestor.');
+  }
```

Comment:
We need to set a default context instead of raising an error as per this issue: https://github.com/openedx/frontend-app-authoring/issues/1510

Reply:
Thanks for catching this, @navinkarkera!
I changed the `LibaryContext` and the `SidebarContext` to return default values here: [829cb5a](https://github.com/openedx/frontend-app-authoring/pull/1539/commits/829cb5abaf4c535f966af4afae8a839e73189931)

It went fine for the `SidebarContext`. For the `LibraryContext`, we will need to handle the case where `libraryId = undefined` at the component level if we don't throw (check the CD/CI errors), and I think it would be better to throw the error at the context level.

What do you think?

Reply:
@rpenido Yes, I think that LibraryContext doesn't need a default, because of that reason - it's annoying to account for libraryId being undefined everywhere, and it doesn't make sense - libraryId is never undefined on any library page. Since our tests already include that context, I think it's better to make it have no default and throw an error in as it was before.

Reply:
Thanks @bradenmacdonald!
Done here: 0de1c076c5325b20cb95fe02b32b367cbf0c9213

Reply:
@rpenido Thanks!